File: Metro/LuckyBallSpirit/DataModel/LotteryInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using LuckyBallsData.Statistics;

namespace LuckyBallSpirit.DataModel
{
    public class BonusInfo
    {
        public string Name { get; set; }
        public string Count { get; set; }
        public string Bonus { get; set; }
    }

    public class LotteryInfo
    {
        public string Index { get; set; }
        public string Issue { get; set; }
        public string Date { get; set; }
        public string Red1 { get; set; }
        public string Red2 { get; set; }
        public string Red3 { get; set; }
        public string Red4 { get; set; }
        public string Red5 { get; set; }
        public string Red6 { get; set; }
        public string Blue { get; set; }
        public string Reds { get; set; }
        public string BetAmount { get; set; }
        public string PoolAmount { get; set; }
        public string More { get; set; }
        public List<BonusInfo> Winners { get; set; }

        public static LotteryInfo Create(Lottery lot)
        {
            LotteryInfo info = new LotteryInfo();

            info.Issue = lot.Issue.ToString();
            info.Date = lot.Date.ToString("yyy-MM-dd");
            info.Red1 = lot.Scheme.Red(0).ToString().PadLeft(2, '0');
            info.Red2 = lot.Scheme.Red(1).ToString().PadLeft(2, '0');
            info.Red3 = lot.Scheme.Red(2).ToString().PadLeft(2, '0');
            info.Red4 = lot.Scheme.Red(3).ToString().PadLeft(2, '0');
            info.Red5 = lot.Scheme.Red(4).ToString().PadLeft(2, '0');
            info.Red6 = lot.Scheme.Red(5).ToString().PadLeft(2, '0');
            info.Blue = lot.Scheme.Blue.ToString().PadLeft(2, '0');
            info.Reds = lot.Scheme.RedsExp;

            info.BetAmount = FormatMoney(lot.BetAmount);
            info.PoolAmount = FormatMoney(lot.PoolAmount);
            info.More = lot.MoreInfo;

            // winners...
            info.Winners = new List<BonusInfo>();

            for (int i = 1; i <= 6; ++i)
            {
                BonusInfo item = new BonusInfo();
                item.Name = i.ToString() + "等奖";
                item.Bonus = FormatMoney(lot.BonusMoney(i));
                item.Count = lot.BonusAmount(i).ToString() + "注";

                info.Winners.Add(item);
            }            

            return info;
        }

        private static string FormatMoney(int money)
        {
            string output = "";
            int yi = money / 100000000;
            if (yi > 0)
                output += yi.ToString() + "亿";

            int wan = (money % 100000000) / 10000;
            if (wan > 0)
                output += wan.ToString() + "万";

            output += (money % 10000).ToString() + "元";

            return output;
        }
    }
}

File: Android/LotterySpirit/src/com/fookwin/lotteryspirit/data/LotteryInfo.java
package com.fookwin.lotteryspirit.data;

import java.text.ParseException;
import java.util.ArrayList;
import java.util.List;

import com.fookwin.lotterydata.data.Lottery;

public class LotteryInfo
{	
    public String Index;
    public String Issue;
    public String Date;
    public String Red1;
    public String Red2;
    public String Red3;
    public String Red4;
    public String Red5;
    public String Red6;
    public String Blue;
    public String Reds;
    public String BetAmount;
    public String PoolAmount;
    public String More;
    public List<BonusInfo> Winners;

    public static LotteryInfo Create(Lottery lot) throws ParseException
    {
        LotteryInfo info = new LotteryInfo();

        info.Issue = Integer.toString(lot.getIssue());
        info.Date = lot.getDateExp();
        info.Red1 = lot.getScheme().getRedExp(0);
        info.Red2 = lot.getScheme().getRedExp(1);
        info.Red3 = lot.getScheme().getRedExp(2);
        info.Red4 = lot.getScheme().getRedExp(3);
        info.Red5 = lot.getScheme().getRedExp(4);
        info.Red6 = lot.getScheme().getRedExp(5);
        info.Blue = lot.getScheme().getBlueExp();
        info.Reds = lot.getScheme().getRedsExp();

        info.BetAmount = FormatMoney(lot.getBetAmount());
        info.PoolAmount = FormatMoney(lot.getPoolAmount());
        info.More = lot.getMoreInfo();

        // winners...
        info.Winners = new ArrayList<BonusInfo>();

        for (int i = 1; i <= 6; ++i)
        {
            BonusInfo item = new BonusInfo();
            item.Name = Integer.toString(i) + "µÈ½±";
            item.Bonus = FormatMoney(lot.BonusMoney(i));
            item.Count = Integer.toString(lot.BonusAmount(i)) + "×¢";

            info.Winners.add(item);
        }            

        return info;
    }

    private static String FormatMoney(int money)
    {
    	String output = "";
        int yi = money / 100000000;
        if (yi > 0)
            output += Integer.toString(yi) + "ÒÚ";

        int wan = (money % 100000000) / 10000;
        if (wan > 0)
            output += Integer.toString(wan) + "Íò";

        output += Integer.toString((money % 10000)) + "Ôª";

        return output;
    }
}

